src/avt/MFEM/avtMFEMDataAdaptor.C

```diff
@@ -405,6 +405,204 @@ avtMFEMDataAdaptor::RefineMeshToVTK(mfem::Mesh *mesh,
+    AVT_MFEM_INFO("Number of Boundary Elements  = " << num_bndry_ele);
+
+    int *mesh_verts_used_by_bndry = new int[num_mesh_vertices];
+    int *mesh_to_bndry_verts_map  = new int[num_mesh_vertices];
+
+    // init mesh_verts_used_by_bndry
+    for (int i=0; i < num_mesh_vertices; i++)
+    {
+        mesh_verts_used_by_bndry[i] = 0;
+        mesh_to_bndry_verts_map[i] = -1;
+    }
+
+    // walk to tag mesh_verts_used_by_bndry
+    for (int i=0; i < num_bndry_ele; i++)
+    {
+        const Element *bndry_ele = mesh->GetBdrElement(i);
+        const int *bndry_ele_verts = bndry_ele->GetVertices();
+
+        for(int j=0;j< bndry_idxs_per_ele;j++)
+        {
+            mesh_verts_used_by_bndry[bndry_ele_verts[j]] = 1;
+        }
+    }
+
+    // count # of boundary verts
+    int num_boundary_verts = 0;
+    for (int i=0; i < num_mesh_vertices; i++)
+    {
+        num_boundary_verts += mesh_verts_used_by_bndry[i];
+    }
+
+    AVT_MFEM_INFO("Number of Boundary Vertices  = " << num_boundary_verts);
+
+    // gen bndry_to_mesh_verts_map and mesh_to_bndry_verts_map
+    int *bndry_to_mesh_verts_map = new int[num_boundary_verts];
+
+    for (int i=0; i < num_boundary_verts; i++)
+    {
+        bndry_to_mesh_verts_map[i] = -1;
+    }
+
+    int bndry_idx = 0;
+    for (int i=0; i < num_mesh_vertices; i++)
+    {
+        if(mesh_verts_used_by_bndry[i] == 1)
+        {
+            bndry_to_mesh_verts_map[bndry_idx] = i;
+            mesh_to_bndry_verts_map[i] = bndry_idx;
+            bndry_idx+=1;
+        }
+    }
+
+    // // if you want to debug these maps
+    // std::cout << " mesh_verts_used_by_bndry = " << std::endl;
+    // for (int i=0; i < num_mesh_vertices; i++)
+    // {
+    //     std::cout << mesh_verts_used_by_bndry[i] << " ";
+    // }
+    // std::cout << endl;
+    //
+    // std::cout << " mesh_to_bndry_verts_map = " << std::endl;
+    // for (int i=0; i < num_mesh_vertices; i++)
+    // {
+    //     std::cout << mesh_to_bndry_verts_map[i] << " ";
+    // }
+    // std::cout << endl;
+    //
+    // std::cout << " bndry_to_mesh_verts_map = "<< std::endl;
+    // for (int i=0; i < num_boundary_verts; i++)
+    // {
+    //     std::cout << bndry_to_mesh_verts_map[i] << " ";
+    // }
+    // std::cout << endl;
+
+    // done with this bookkeeping array
+    delete [] mesh_verts_used_by_bndry;
+
+    ////////////////////////////////////////////
+    // Setup bndry points
+    ////////////////////////////////////////////
+
+    vtkPoints *points = vtkPoints::New();
+    points->SetDataTypeToDouble();
+    points->SetNumberOfPoints(num_boundary_verts);
+
+    GridFunction *mesh_gf = mesh->GetNodes();
+
+    if(mesh_gf == NULL)
+    {
+        AVT_MFEM_INFO("Mesh does not have Nodes Grid Function ");
+    }
+    else
+    {
+        AVT_MFEM_INFO("Mesh does have Nodes Grid Function ");
+    }
+
+    //-------------------------------------------------------//
+    // TODO/NOTE: There may be cases where the vertices exist
```

Comment:
Did you mean to leave the TODO? Future work?

Reply:
This would be future work, I'll add a better note.

Reply:
Added note